assignment_2_Script.js

```diff
@@ -0,0 +1,81 @@
+var dateCreation = (function iife() {
+    var currentModuleDate = null;
+    function instantiateDate() {
+        return new Date();
+    }
+
+    function setDate(date) {
+        if (date===undefined) {
+            currentModuleDate = instantiateDate();
+        }
+        else if (typeof date === 'number') {
+            currentModuleDate = new Date(date);
+        }
+        else if (typeof date === 'object') {
```

Comment:
Two things:

1. using typeof here allows anything that reads as an "object" to be passed in:

    `dateCreation.setDate([]);`

for example, will pass the logic check, but then break your other methods as an array literal is not a valid Date object. Instead, check out the documentation for instanceof

2. don't keep a reference to the passed-in date - you can instantiate a new date instance by simply passing in the reference to the Date constructor, rather than storing that reference as your internal reference. Keeping the external reference means external code can modify or even overwrite your internal reference.